cordyceps/src/sorted_list.rs

```diff
@@ -0,0 +1,459 @@
+//! [Intrusive], singly-linked, sorted, linked-list.
+//!
+//! See the documentation for the [`SortedList`] and [`SortedListIter`] types for
+//! details.
+//!
+//! [intrusive]: crate#intrusive-data-structures
+
+use crate::{Linked, Stack};
+use core::{marker::PhantomData, ptr::NonNull};
+
+pub use crate::stack::Links;
+
+/// A sorted singly linked list
+///
+/// This behaves similar to [`Stack`], in that it is a singly linked list,
+/// however items are stored in an ordered fashion. This means that insertion
+/// is an `O(n)` operation, and retrieval of the first item is an `O(1)` operation.
```

Comment:
nit: i would format this as:
```suggestion
/// is an _O_(_n_) operation, and retrieval of the first item is an _O_(1) operation.
```